patches/chromium/process_singleton.patch

```diff
@@ -279,7 +290,16 @@ index e2312c65623a3f54ecbc18058720ccae53822e76..ec725b44296266bea1a51aea889463a0
        is_virtualized_(false),
        lock_file_(INVALID_HANDLE_VALUE),
        user_data_dir_(user_data_dir),
-@@ -368,13 +379,16 @@ ProcessSingleton::NotifyOtherProcessOrCreate() {
+       should_kill_remote_process_callback_(
+-          base::BindRepeating(&DisplayShouldKillMessageBox)) {}
++          base::BindRepeating(&DisplayShouldKillMessageBox)) {
++  // The user_data_dir may have not been created yet.
++  base::CreateDirectoryAndGetError(user_data_dir, nullptr);
```

Comment:
I see this is based on the previous code, instead of having to check for the directory in both the windows and linux implementation, it can instead be performed at the single entry point before creating the `ProcessSingleton` instance https://github.com/electron/electron/blob/main/shell/browser/api/electron_api_app.cc#L1149-L1150